gen.go

```diff
@@ -1043,8 +1044,8 @@ func (t *{{ .Name}}) UnmarshalCBOR(r io.Reader) (err error) {
 		return err
 	}
 	defer func() {
-		if err == io.EOF {
-			err = io.ErrUnexpectedEOF
+		if errors.Is(err, io.EOF) {
+			err = xerrors.Errorf("%w: %v", io.ErrUnexpectedEOF, err)
```

Comment:
This doesn't quite work. `%w` needs to be last.

It's _probably_ fine to just leave it as-is? The only thing we _really_ care about is not returning an EOF if we've read bytes.

Reply:
Does it? https://go.dev/play/p/k2wOyoCk4WO makes it seem like both work.

I figured we cared enough to wrap the original error, so might be nice to keep that info.

Reply:
Ah, being last only matters for `Unwrap`. (docs: https://pkg.go.dev/golang.org/x/xerrors#Errorf)

> I figured we cared enough to wrap the original error, so might be nice to keep that info.

My point was: that we could just leave everything as-is. I.e., we _don't_ want an EOF unless we read no data, but we're fine with any other error (including a wrapped EOF) otherwise.

But if this works and doesn't hurt performance too much, then it's fine by me.

Reply:
> Ah, being last only matters for Unwrap. (docs: https://pkg.go.dev/golang.org/x/xerrors#Errorf)

From that link
> If the format specifier includes a %w verb with an error operand in a position other than at the end, the returned error will still implement an Unwrap method returning the operand, but the error's Format method will not return the wrapped error. 

What is the "error's Format method" ?

anyways, I understand what you mean by as-is. I changed this to return the wrapped error and only replace if we return EOF. I also changed the test to only check that we aren't returning EOF exactly if it could read some bytes.